packages/test/test-end-to-end-tests/src/test/gc/gcAttachmentBlobs.spec.ts

```diff
@@ -320,47 +340,53 @@ describeNoCompat("Garbage collection of blobs", (getTestObjectProvider) => {
+            // Upload the same blob. This will get de-duped and we will get back a handle with another localId. This and
+            // the blob uploaded in disconnected mode should map to the same storageId.
+            const localHandle2 = await defaultDataStore._context.uploadBlob(stringToBuffer(blobContents, "utf-8"));
 
-            // Delete the localId handle. The localId and storageId nodes should both be unreferenced. They will be
-            // be deleted locally in this summary. In the next summary, they will be deleted.
-            container2DataStore._root.delete("local");
+            // Add the localId2 handle and remove the localId1 handle. Validate that localId2 and storageId nodes are
+            // referenced and localId1 node is unreferenced.
+            defaultDataStore._root.set("local2", localHandle2);
+            defaultDataStore._root.delete("local1");
             const s2 = await summarizeAndGetUnreferencedNodeStates(summarizerRuntime);
-            assert(s2.get(localHandle.absolutePath) === "unreferenced", "local id blob should be unreferenced");
-            assert(s2.get(storageHandle.absolutePath) === "unreferenced", "storage id blob should be unreferenced");
+            assert(s2.get(localHandle1.absolutePath) === "unreferenced", "local id 1 blob should be unreferenced");
+            assert(s2.get(localHandle2.absolutePath) === "referenced", "local id 2 blob should be referenced");
+            assert(s2.get(storageId) === "referenced", "storage id blob should also be referenced");
 
-            // In this summary, unreferenced nodes will be deleted if deleteUnreferencedContent is set.
+            // Remove the localId2 handle. Validate that localId2 and storageId nodes are now unreferenced. Also, if
+            // deleteUnreferencedContent is true, localId1 node would have been deleted.
+            defaultDataStore._root.delete("local2");
             const s3 = await summarizeAndGetUnreferencedNodeStates(summarizerRuntime);
             if (deleteUnreferencedContent) {
-                assert(s3.get(localHandle.absolutePath) === undefined, "local id blob should not have a GC entry");
-                assert(s3.get(storageHandle.absolutePath) === undefined, "storage id blob should not have a GC entry");
+                assert(s3.get(localHandle1.absolutePath) === undefined, "local id 1 blob should not have a GC entry");
             } else {
-                assert(s3.get(localHandle.absolutePath) === "unreferenced", "local id blob should be unreferenced");
-                assert(s3.get(storageHandle.absolutePath) === "unreferenced", "storage id blob should be unreferenced");
+                assert(s3.get(localHandle1.absolutePath) === "unreferenced", "local id 1 blob should still be unreferenced");
             }
+            assert(s3.get(localHandle2.absolutePath) === "unreferenced", "local id 2 blob should still be unreferenced");
```

Comment:
Why is this unreferenced regardless of `deleteUnreferencedContent`?

Reply:
Nodes are deleted _after_ we generate the GC data and run GC. So, these nodes are present in the GC data of this run. This is removed from the GC data in the next run.
We cannot remove it from the current GC data without re-running GC becuase we cannot be sure what objects are going to do when asked to delete sweepReady nodes. It may be worth evaluating them during the sweep work.

Reply:
Now that I look at this again, I'm ok with us having unreferenced content in a summary before we delete it. It seems that localHandle2 was referenced in s2 and unreferenced in s3. A slower delete is much safer than a fast delete. Feel free to resolve.

Reply:
It's fine for the time being. But it way pose issues when we enable sweep if sweep has similar semantics. Can you create a task with this so we are tracking it for the sweep work.